src/toil/lib/retry.py

```diff
@@ -572,6 +572,7 @@ def old_retry(
     >>> i
     1
     """
+    timeout = timeout if timeout else DEFAULT_TIMEOUT
```

Comment:
Won't this break `timeout=0`? Maybe this should be:
```suggestion
    if timeout is None:
        timeout = DEFAULT_TIMEOUT
```

Or maybe we really want `timeout=None` to represent not using a timeout?

Reply:
I think setting it to a default might be better. With the current retry code the only way to simply set an infinite timeout is to just set the timeout to a really high number.